lib/powerpoint/slide/table_cell.rb

```diff
@@ -0,0 +1,210 @@
+require 'erb'
+
+module Powerpoint
+  module Slide
+    class TableCell
+      include Powerpoint::Util
+
+      attr_reader :horizontal_size, :vertical_size, :data, :right, :bottom
+
+      def initialize(data)
+        @data = data
+        @right = []
+        @bottom = []
+      end
+
+      def to_renderable(go_down = true, template = read_template('_table_cell.xml.erb'))
+
+        horizontal, vertical = calculate_sizing
+
+        cells =
+            (0..vertical-1).map do |i|
+              (0..horizontal-1).map do |j|
+
+                options = {}
+                if i == 0 && j == 0
+                  options[:gridspan] = horizontal >= 2 ? horizontal : false
+                  options[:rowspan] = vertical >= 2 ? vertical : false
+                  options[:data] = @data
+
+                elsif i == 0
+                  options[:h_merge] = 1
+                  options[:rowspan] = vertical >= 2 ? vertical : false
+                  options[:template] = template
+                elsif j == 0
+                  options[:gridspan] = horizontal >= 2 ? horizontal : false
+                  options[:v_merge] = i
+                  options[:template] = template
+                else
+                  options[:h_merge] = 1
+                  options[:v_merge] = i
+                  options[:template] = template
+                end
+
+                options[:template] = template
+
+                RenderableCell.new(options)
+              end
+            end
+
+        neighbor_rows = @right.flat_map { |n| n.to_renderable(false, template) }
+
+        # join right neighbor rows with mine if we have neighbors
+        cells = cells.zip(neighbor_rows).map { |myrow, nrow| myrow + nrow } unless neighbor_rows.empty?
+
+        return cells if @bottom.empty? || !go_down
+
+        bot_rows = @bottom[0].to_renderable(true, template)
+        # puts "Bottom: #{bot_rows}"
+
+        cells + bot_rows
+      end
+
+      def set_right_neighbor(neighbor)
+        @right = [neighbor]
+      end
+
+      def set_right_neighbors(neighbors)
```

Comment:
This can be simplified to just wrap the argument with `Array` ie `Array(neighbors)`. It will not double wrap an array and will wrap a single argument.